### DISCUSSION TITLE
Is there a way to get Credentials from aws_config?

### DISCUSSION THREAD

**Discussion by allan2** [awslabs/aws-sdk-rust#268] [Q&A] [answered]
Is there a way to get `aws_types::Credentials` from `aws_config::config::Config`?

```rust
let shared_config = aws_config::from_env().region(region).load().await;

let creds = aws_types::Credentials::new(
    dotenv!("AWS_ACCESS_KEY_ID"),      // <--- get this from shared_config
    dotenv!("AWS_SECRET_ACCESS_KEY"),  // <--- get this from shared_config
    None,
    None,
    "provider_name",
);
```

**Comment by rcoh** [comment, marked as answer]
you can retrieve them from the credentials provider: https://docs.rs/aws-types/0.0.22-alpha/aws_types/config/struct.Config.html#method.credentials_provider

```rust
let shared_config = aws_config::from_env().region(region).load().await;
let creds = shared_config
  .credentials_provider().expect("a credentials provider is loaded")
  .provide_credentials().await
```

**Reply by allan2** [reply]
Can I still set expiration time and provider name?

**Reply by rcoh** [reply]
I'm a little confused by what you're trying to do, can you explain your entire use case?

**Reply by allan2** [reply]
I'm passing `Credentials` to `aws_sig_auth` for RDS.
`Credentials` has `expires_after` and `provider_name` fields that I would like to set.
I'm wondering how I can do that if I am creating my credentials from a `Config`.

**Reply by rcoh** [reply]
Ah that makes sense! You should use the [DefaultCredentialsChain](https://awslabs.github.io/aws-sdk-rust/aws_config/default_provider/credentials/struct.DefaultCredentialsChain.html) directly.

When you call `provide_credentials().await` on the default credentials chain it will return credentials you can use with RDS. You don't need to manually construct your own credentials object.
